### PR TITLE
feat: add `mongodb-setup` example

### PR DESCRIPTION
**Pull request by mateosantosdev**

### CODE REVIEW

File: mongodb-setup/.prettierrc
Comment (MichaelDeBoey) [inline mongodb-setup/.prettierrc:1]: This file shouldn't be included

File: mongodb-setup/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# MongoDB support
+
+Connection and user login with a MongoDB database.
+
+This project uses the future [route v2 convention](https://remix.run/docs/en/main/pages/v2#file-system-route-convention).
```
Comment (MichaelDeBoey) [inline mongodb-setup/README.md:5]: Please revert to v1 routing convention, so it's in line with the rest of the examples

File: mongodb-setup/README.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+5. Go to `http://localhost:3000/login`
+
+### User collection schema
+
+```
+email: string
+password: string 
+```
+
+The password is hashed with the help of the bcryptjs library
```
Comment (MichaelDeBoey) [inline mongodb-setup/README.md:38]:
```suggestion
The password is hashed with the help of the `bcryptjs` library
```

File: mongodb-setup/app/routes/api.create-demo-user.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import { UserApi } from "~/user/api";
+export async function loader() {
```
Comment (MichaelDeBoey) [inline mongodb-setup/app/routes/api.create-demo-user.ts:2]: Keeping this consistent with the rest of the examples

```suggestion
export const loader = async () => {
```

File: mongodb-setup/app/routes/api.create-demo-user.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+  const userService = new UserApi();
+  await userService.insertDemo("youruseremail@demo.com");
+  const response: any = await userService.getAllUsers();
+  return response;
```
Comment (MichaelDeBoey) [inline mongodb-setup/app/routes/api.create-demo-user.ts:6]:
```suggestion

  return userService.getAllUsers();
```

File: mongodb-setup/app/server/db.server.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+            db = (global as any).__db;
+        }
+        return db;
+    } catch (error: any) {
+        console.error('--- ERROR', error);
+    }
+}
+
+// Users
+const UserSchema: Schema = new Schema<UserModel>({
```
Comment (MichaelDeBoey) [inline mongodb-setup/app/server/db.server.ts:30]: No need for explicit typing here

```suggestion
const UserSchema = new Schema<UserModel>({
```

File: mongodb-setup/app/types/index.ts [hunk 6]
```diff
@@ -0,0 +1,9 @@
+export interface UserModel extends Document {
+  email: string;
+  password: string;
+}
```
Comment (MichaelDeBoey) [inline mongodb-setup/app/types/index.ts:4]:
```suggestion
export type UserModel = Document & {
  email: string;
  password: string;
}
```

File: mongodb-setup/remix.config.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+  // assetsBuildDirectory: "public/build",
+  // serverBuildPath: "build/index.js",
+  // publicPath: "/build/",
+  serverModuleFormat: "cjs",
+  future: {
+    v2_errorBoundary: true,
+    v2_meta: true,
+    v2_normalizeFormMethod: true,
+    v2_routeConvention: true,
+  },
```
Comment (MichaelDeBoey) [inline mongodb-setup/remix.config.js:14]: Keeping this consistent with the other examples

```suggestion
```

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]

File: mongodb-setup/app/routes/login.tsx [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+    const fields = {
+      username: username.toString(),
+      password: password.toString(),
+    };
+
+    const user = await login(fields);
+    if (!user) throw new Error("Username not found");
+    return createUserSession(user.id, user.email, "/");
+  } catch (error: any) {
```
Comment (MichaelDeBoey) [inline mongodb-setup/app/routes/login.tsx:24]:
```suggestion
  } catch (error: unknown) {
```

File: mongodb-setup/remix.config.js [hunk 9]
```diff
@@ -0,0 +1,8 @@
+/** @type {import('@remix-run/dev').AppConfig} */
+module.exports = {
+  ignoredRouteFiles: ["**/.*"],
+  // appDirectory: "app",
+  // assetsBuildDirectory: "public/build",
+  // serverBuildPath: "build/index.js",
+  // publicPath: "/build/"
```
Comment (MichaelDeBoey) [inline mongodb-setup/remix.config.js:7]:
```suggestion
  // publicPath: "/build/",
```

File: mongodb-setup/app/server/session.server.ts [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+    const id = session.get('id');
+    if (!id || typeof id !== 'string') {
+        throw redirect(`/login`);
+    }
+    return id;
+}
+
+export async function login({ username, password }: LoginForm) {
+    const user = await userService.getUserByEmail(username);
+    console.log('Usuario', user);
```
Comment (MichaelDeBoey) [inline mongodb-setup/app/server/session.server.ts:65]:
```suggestion
```

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]
@mateosantosdev Please also fix ESLint errors
https://github.com/remix-run/examples/actions/runs/5381830562/jobs/9771435870?pr=273

If you run `yarn lint:fix` in the root of the repo, these should all be fixed

**Comment by mateosantosdev** [comment]
> @mateosantosdev Please also fix ESLint errors https://github.com/remix-run/examples/actions/runs/5381830562/jobs/9771435870?pr=273
> 
> If you run `yarn lint:fix` in the root of the repo, these should all be fixed

Thank you @MichaelDeBoey , the eslint issues have been fixed

**Comment by mateosantosdev** [comment]
Hello @MichaelDeBoey I fixed the lint errors the last week.

Do you need anything else?

Thank you in advance!

File: mongodb-setup/package.json [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+    "isbot": "^3.6.5",
+    "mongoose": "^7.3.1",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0"
+  },
+  "devDependencies": {
+    "@remix-run/dev": "~1.14.2",
+    "@remix-run/eslint-config": "~1.14.2",
+    "@types/bcryptjs": "^2.4.2",
+    "@types/mongoose": "^5.11.97",
```
Comment (MichaelDeBoey) [inline mongodb-setup/package.json:25]: `mongoose` comes with it's own types, so no need to install them from DT anymore

```suggestion
```

File: mongodb-setup/app/server/db.server.ts [hunk 12]
```diff
@@ -0,0 +1,38 @@
+import * as dotenv from 'dotenv';
```
Comment (MichaelDeBoey) [inline mongodb-setup/app/server/db.server.ts:1]: There's a shorter import for doing this

```suggestion
import from 'dotenv/config';
```

File: mongodb-setup/app/server/db.server.ts [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+import mongoose, { Schema } from 'mongoose';
+
+import type { UserModel } from '~/types';
+
+dotenv.config();
+
```
Comment (MichaelDeBoey) [inline mongodb-setup/app/server/db.server.ts:7]:
```suggestion
```

File: mongodb-setup/app/server/session.server.ts [hunk 14]
```diff
@@ -0,0 +1,79 @@
+import { createCookieSessionStorage, redirect } from '@remix-run/node';
+import bcrypt from 'bcryptjs';
+import * as dotenv from 'dotenv';
```
Comment (MichaelDeBoey) [inline mongodb-setup/app/server/session.server.ts:3]:
```suggestion
import from 'dotenv/config';
```

File: mongodb-setup/app/server/session.server.ts [hunk 15]
```diff
@@ -0,0 +1,79 @@
...
+
+import type { LoginForm } from '~/types';
+import { UserApi } from '~/user/api';
+dotenv.config();
```
Comment (MichaelDeBoey) [inline mongodb-setup/app/server/session.server.ts:7]:
```suggestion
```

File: mongodb-setup/app/server/session.server.ts [hunk 16]
```diff
@@ -0,0 +1,79 @@
...
+    const session = await storage.getSession();
+    session.set('id', id);
+    session.set('email', email);
+    return redirect(redirectTo, {
+        headers: {
+            'Set-Cookie': await storage.commitSession(session)
+        }
+    });
+}
+function getUserSession(request: Request) {
```
Comment (MichaelDeBoey) [inline mongodb-setup/app/server/session.server.ts:37]: Let's make it more clear that this is an async function

```suggestion
async function getUserSession(request: Request) {
```

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]

File: mongodb-setup/app/server/db.server.ts (hunk 12 above)
Comment (mateosantosdev) [reply mongodb-setup/app/server/db.server.ts:1]: Hello @MichaelDeBoey , I receive this error

error importing

**Comment by mateosantosdev** [comment]
I have closed this PR due to inactivity.

**Pull request closed without merging**